File: tests/XFAIL/base64_encoding/main.go
package main

import (
	"encoding/base64"
	"fmt"
)

func main() {
	data := "Hello, World!"
	encoded := base64.StdEncoding.EncodeToString([]byte(data))
	fmt.Println("Encoded:", encoded)

	decoded, _ := base64.StdEncoding.DecodeString(encoded)
	fmt.Println("Decoded:", string(decoded))
}

File: tests/XFAIL/base64_encoding/main.rs
use std::cell::{RefCell};
use std::rc::{Rc};

fn main() {
    let mut data = Rc::new(RefCell::new(Some("Hello, World!".to_string())));
    let mut encoded = (*base64::std_encoding.borrow().as_mut().unwrap()).encode_to_string(Rc::new(RefCell::new(Some(Rc::new(RefCell::new(Some((*data.borrow().as_ref().unwrap()).as_bytes().to_vec())))))));
    println!("{} {}", "Encoded:".to_string(), (*encoded.borrow().as_ref().unwrap()));

    let (mut decoded, _) = (*base64::std_encoding.borrow().as_mut().unwrap()).decode_string(Rc::new(RefCell::new(Some((*encoded.borrow().as_ref().unwrap())))));
    println!("{} {}", "Decoded:".to_string(), (*Rc::new(RefCell::new(Some(String::from_utf8((*decoded.borrow().as_ref().unwrap()).clone()).unwrap()))).borrow().as_ref().unwrap()));
}